dask/dataframe/io/parquet/arrow.py

```diff
@@ -0,0 +1,344 @@
+        if df.index.name:
+            df = df.reset_index()
+
+        return df[list(columns)]
+
+    @staticmethod
+    def read_metadata(
+        fs, paths, categories=None, index=None, gather_statistics=None, **kwargs
+    ):
+
+        # In pyarrow, the physical storage field names may differ from
+        # the actual dataframe names. This is true for Index names when
+        # PyArrow >= 0.8.
+        # We would like to resolve these to the correct dataframe names
+        # as soon as possible.
+
+        dataset = pq.ParquetDataset(
+            paths, filesystem=get_pyarrow_filesystem(fs)
+        )
+
+        if dataset.partitions is not None:
+            partitions = [
+                n for n in dataset.partitions.partition_names if n is not None
+            ]
+        else:
+            partitions = []
+
+        schema = dataset.schema.to_arrow_schema()
+        columns = None
+
+        has_pandas_metadata = (
+            schema.metadata is not None and b"pandas" in schema.metadata
+        )
+
+        if has_pandas_metadata:
+            pandas_metadata = json.loads(
+                schema.metadata[b"pandas"].decode("utf8")
+            )
+            index_names, column_names, storage_name_mapping, column_index_names = _parse_pandas_metadata(
+                pandas_metadata
+            )
+        else:
+            index_names = []
+            column_names = schema.names
+            storage_name_mapping = {k: k for k in column_names}
+            column_index_names = [None]
+
+        if index is None and len(index_names) > 0:
+            if len(index_names) == 1:
+                index = index_names[0]
+            else:
+                index = index_names
+
+        column_names += [p for p in partitions if p not in column_names]
+        column_names, index_names = _normalize_index_columns(
+            columns, column_names, index, index_names
+        )
+
+        all_columns = index_names + column_names
+
+        pieces = sorted(
+            dataset.pieces, key=lambda piece: natural_sort_key(piece.path)
+        )
+
+        # Check that categories are included in columns
+        if categories and not set(categories).intersection(all_columns):
+            raise ValueError(
+                "categories not in available columns.\n"
+                "categories: {} | columns: {}".format(
+                    categories, list(all_columns)
+                )
+            )
+
+        dtypes = _get_pyarrow_dtypes(schema, categories)
+        dtypes = {storage_name_mapping.get(k, k): v for k, v in dtypes.items()}
+
+        meta = _meta_from_dtypes(
+            all_columns,
+            dtypes,
+            index_cols=(),
+            column_index_names=column_index_names,
+        )
+
+        meta = clear_known_categories(meta, cols=categories)
+        if gather_statistics is None and dataset.metadata:
+            gather_statistics = True
+
+        if gather_statistics and pieces:
```

Comment:
Are you suggesting that we remove `gather_statistics` as an argument?  By default, the only way this will *not* be done is if the metadata is not availabale (or if `gather_statistics` is explicitly set to `False`)

Reply:
No, I'm thinking that maybe it should only be optional in he case the metadata file is missing. Maybe defer this conversation - we prefer the metadata file every time for now.